src/community/features-templating/features-templating-ows/src/test/java/org/geoserver/featurestemplating/wfs/GetCapabilitiesTest.java

```diff
@@ -0,0 +1,42 @@
+/* (c) 2023 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.featurestemplating.wfs;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.Set;
+import java.util.TreeSet;
+import org.geoserver.wfs.WFSTestSupport;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+/** GetCapabilities test for features templating module */
+public class GetCapabilitiesTest extends WFSTestSupport {
+
+    /**
+     * Checks if features templating specific result formats are present in output and represented
+     * correctly
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testOutputFormats() throws Exception {
+        Document doc = getAsDOM("wfs?service=WFS&request=getCapabilities&version=1.0.0");
+
+        Element outputFormats = getFirstElementByTagName(doc, "ResultFormat");
+        NodeList formats = outputFormats.getChildNodes();
+
+        Set<String> s1 = new TreeSet<>();
+        for (int i = 0; i < formats.getLength(); i++) {
+            String format = formats.item(i).getNodeName();
+            s1.add(format);
+        }
+
+        assertTrue(s1.contains("JSON-LD"));
+        assertTrue(s1.contains("HTML"));
+    }
```

Comment:
Makes sense for WFS... should we have the same for WMS too (in another test eventually)? The GetFeatureInfo description in GetCapabilities also provides a list of formats.

Reply:
Will discuss this